### PR TITLE
Migrate `PatientEntryScreen` to use view effects

### PR DESCRIPTION
**Pull request by msasikanth**
https://app.shortcut.com/simpledotorg/story/5691/migrate-patiententryscreen-to-use-view-effects

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/newentry/PatientEntryViewEffectHandler.kt [hunk 1]
```diff
@@ -53,4 +55,36 @@ class PatientEntryViewEffectHandler(
       showDOBExceedsMinLimitError(false)
     }
   }
+
+  private fun showValidationErrors(errors: List<PatientEntryValidationError>) {
+    errors
+        .onEach { Analytics.reportInputValidationError(it.analyticsName) }
+        .forEach {
+          when (it) {
+            PatientEntryValidationError.FullNameEmpty -> validationActions.showEmptyFullNameError(true)
```
Comment (Honey14) [inline app/src/main/java/org/simple/clinic/newentry/PatientEntryViewEffectHandler.kt:64]: Why do we have a different interface for `PatientEntryValidationActions` (UiActions)? It should ideally be included in the `PatientEntryUiActions`. WDYT?

**Review by Honey14 — CHANGES_REQUESTED** [review]
Looks good. left one comment

File: app/src/main/java/org/simple/clinic/newentry/PatientEntryViewEffectHandler.kt (hunk 1 above)
Comment (msasikanth) [reply app/src/main/java/org/simple/clinic/newentry/PatientEntryViewEffectHandler.kt:64]: Yep, good point. Will make that change.

**Review by Honey14 — APPROVED** [review]
LGTM

**Pull request merged**